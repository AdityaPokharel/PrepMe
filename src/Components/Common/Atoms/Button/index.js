import React from 'react';
import PropTypes from 'prop-types';
import {Text, TouchableOpacity, StyleSheet} from 'react-native';

const Button = ({label, onPress, extendedStyles}) => {
  const styles = styleSheet();
  return (
    <TouchableOpacity
      activeOpacity={0.9}
      style={{
        ...styles.wrapper,
        ...extendedStyles,
      }}
      onPress={onPress}>
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  extendedStyles: PropTypes.object,
};

Button.defaultProps = {
  extendedStyles: {},
};

const styleSheet = theme =>
  StyleSheet.create({
    wrapper: {
      width: '100%',
      height: 48,
      backgroundColor: '#4d13da',
      justifyContent: 'center',
      borderRadius: 5,
    },
    text: {
      textAlign: 'center',
      color: 'white',
      fontWeight: 'bold',
    },
  });

export default Button;
