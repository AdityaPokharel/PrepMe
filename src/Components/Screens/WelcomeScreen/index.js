import React from 'react';
import {Image, SafeAreaView, StyleSheet, Text, View} from 'react-native';
import Button from '../../Common/Atoms/Button';
import Heading1 from '../../Common/Atoms/Heading1';
import LightText from '../../Common/Atoms/LightText';

const t = {
  welcome: {
    title: 'Welcome to your preparation tool.',
    text:
      'PrepMe helps your squeeze in more content in small, bite-sized chunks.',
  },
};

const WelcomeScreen = ({imageDimensions, onLayout, onPress}) => {
  const styles = styleSheet(imageDimensions);
  return (
    <SafeAreaView style={styles.mainWrapper}>
      <View style={styles.wrapper}>
        <View style={styles.textWrapper}>
          <Heading1>{t.welcome.title}</Heading1>
          <LightText extendedStyles={styles.text}>{t.welcome.text}</LightText>
        </View>
        <Image
          onLayout={onLayout}
          style={imageDimensions}
          source={require('../../../assets/images/welcome-image.png')}
        />
        <View style={styles.buttonWrapper}>
          <Button onPress={onPress} label="NEXT" />
        </View>
      </View>
    </SafeAreaView>
  );
};

const styleSheet = theme =>
  StyleSheet.create({
    mainWrapper: {
      backgroundColor: '#f5f5f5',
    },
    wrapper: {
      alignItems: 'center',
      justifyContent: 'space-between',
      height: '100%',
    },
    textWrapper: {
      width: '95%',
      marginTop: 15,
    },
    imageWrapper: {
      width: '100%',
    },
    text: {
      marginTop: 20,
    },
    image: {
      width: theme.width,
      height: theme.height,
    },
    buttonWrapper: {
      width: '95%',
      marginBottom: 10,
    },
  });

export default WelcomeScreen;
