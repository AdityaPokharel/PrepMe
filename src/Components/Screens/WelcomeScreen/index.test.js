import React from 'react';
import {fireEvent, render} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import WelcomeScreen from './index';
import Button from '../../Common/Atoms/Button';

describe('welcome screen tests', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <WelcomeScreen
        onPress={() => {}}
        imageDimensions={{width: 30, height: 30}}
        onLayout={() => {}}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should press onPress on button press', () => {
    const onPress = jest.fn();
    const setDimensions = jest.fn();
    const {getByType} = render(
      <WelcomeScreen
        onPress={onPress}
        imageDimensions={{width: 30, height: 30}}
        onLayout={setDimensions}
      />,
    );
    expect(onPress).toHaveBeenCalledTimes(0);
    const button = getByType(Button);
    fireEvent.press(button);
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
