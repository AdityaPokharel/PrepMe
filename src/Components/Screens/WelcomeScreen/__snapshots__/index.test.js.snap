// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome screen tests should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 15,
          "width": "95%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1A1A1A",
            "fontSize": 32,
            "fontWeight": "500",
          }
        }
      >
        Welcome to your preparation tool.
      </Text>
      <Text
        style={
          Object {
            "color": "#666666",
            "marginTop": 20,
          }
        }
      >
        PrepMe helps your squeeze in more content in small, bite-sized chunks.
      </Text>
    </View>
    <Image
      onLayout={[Function]}
      source={
        Object {
          "testUri": "../../../src/assets/images/welcome-image.png",
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <View
      style={
        Object {
          "marginBottom": 10,
          "width": "95%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#4d13da",
            "borderRadius": 5,
            "height": 48,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          NEXT
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
